### ISSUE TITLE
Enable resource manager metrics by default

### ISSUE THREAD

**Issue by sukunrt** [libp2p/go-libp2p#2372]
We have found these metrics very useful and the cost of reporting metrics is not a lot. Users can opt out if they want to. In line with the rest of the places in libp2p we should enable these metrics by default too.

**Comment by marten-seemann** [comment]
Sounds good to me, let's do it!

**Comment by sukunrt** [comment]
The current implementation of TraceReporter is in a separate package `github.com/libp2p/go-libp2p/p2p/host/resource-manager/obs` which depends on `rcmgr` package. 

So from within rcmgr there is no way to initialise a new `StatsTraceReporter`. 

There are two options here:
1. Move `obs.StatsTraceReporter` to the same package. This would be a breaking change
2. Add a methods like Init on `rcmgr` which is called while setting up BasicHost

I lean towards 1 since users using `StatsTraceReporter` don't need to pass it explicity after this any way.

**Comment by MarcoPolo** [comment]
Can you do 1 without a breaking change?  I.e both ways work

**Comment by sukunrt** [comment]
@MarcoPolo I think I can. Can you check #2388 and #2389 ? 

I've moved the type `StatsTraceReporter` to `rcmgr` package and created an alias in package `obs` to the this `rcmgr` type like 

```
package obs 

type StatsTraceReporter = rcmgr.StatsTraceReporter
```

kubo works fine with this. I think this should be fine.

**Comment by marten-seemann** [comment]
> Can you do 1 without a breaking change? I.e both ways work

Why? This is only used by Kubo and Lotus (and maybe a very small number of consumers). I'd rather do the right thing, instead of creating additional API surface for the sake of backwards compatibility.

**Comment by MarcoPolo** [comment]
> Why? This is only used by Kubo and Lotus (and maybe a very small number of consumers). I'd rather do the right thing, instead of creating additional API surface for the sake of backwards compatibility.

I'd argue that the right thing is avoiding unnecessary breakages for users. They've already jumped through the hoops of setting this up once, let's not force a harder upgrade for them.

**Comment by marten-seemann** [comment]
Building in backwards compatibility comes with a long-term maintenance cost for us. Since we're creating the PRs for Kubo and Lotus anyway, there's no cost for them to pay in the first place.

If we want to avoid breaking users, we should at least deprecate the option that we don't want people to use any more.

**Comment by marten-seemann** [comment]
#2388 deprecated the obs package, so this should be fine.

**Comment by marten-seemann** [comment]
Fixed by #2389.
